runtime/cmd/decode-state-values/main.go

```diff
@@ -258,111 +259,137 @@ func load() {
+	if isSlabStorageKey(key) {
 
-				storageID := atree.StorageID{
-					Address: address,
-					Index:   storageIndex,
-				}
+		// Only decode each slab if it was not already decoded
+		// for the slab health check
 
-				_, err := decodeSlab(storageID, data)
-				if err != nil {
-					log.Printf(
-						"Failed to decode slab @ %s: %s (size: %d)",
-						storageID, err, len(data),
-					)
-					continue
-				}
+		if !*checkSlabsFlag {
+
+			var storageIndex atree.StorageIndex
+			// Skip '$' prefix
+			copy(storageIndex[:], key[1:])
+
+			storageID := atree.StorageID{
+				Address: address,
+				Index:   storageIndex,
 			}
-		} else {
-			// If the key is an account path,
-			// decode the storable, and load the value
-
-			keyParts := strings.SplitN(key, storagePathSeparator, 2)
-
-			isStoragePath = len(keyParts) == 2 &&
-				common.PathDomainFromIdentifier(keyParts[0]) != common.PathDomainUnknown
-
-			if isStoragePath {
-
-				reader := bytes.NewReader(data)
-				decoder := interpreter.CBORDecMode.NewStreamDecoder(reader)
-				storable, err := interpreter.DecodeStorable(decoder, atree.StorageIDUndefined)
-				if err != nil {
-					log.Printf(
-						"Failed to decode storable @ 0x%x %s: %s (data: %x)",
-						address, key, err, data,
-					)
-					continue
-				}
 
-				atreeValue, err := storable.StoredValue(slabStorage)
-				if err != nil {
-					log.Printf(
-						"Failed to load stored value @ 0x%x %s: %s",
-						address, key, err,
-					)
-					continue
-				}
+			_, err := decodeSlab(storageID, data)
+			if err != nil {
+				log.Printf(
+					"Failed to decode slab @ %s: %s (size: %d)",
+					storageID, err, len(data),
+				)
+				return err
+			}
+		}
+	} else {
+		// If the key is an account path,
+		// decode the storable, and load the value
 
-				value, err := interpreter.ConvertStoredValue(atreeValue)
-				if err != nil {
-					log.Printf(
-						"Failed to convert stored value @ 0x%x %s: %s",
-						address, key, err,
-					)
-					continue
-				}
+		keyParts := strings.SplitN(key, storagePathSeparator, 2)
 
-				interpreter.InspectValue(
-					value,
-					func(v interpreter.Value) bool {
+		isStoragePath := len(keyParts) == 2 &&
+			common.PathDomainFromIdentifier(keyParts[0]) != common.PathDomainUnknown
 
-						if composite, ok := v.(*interpreter.CompositeValue); ok &&
-							composite.Kind == common.CompositeKindResource {
+		if isStoragePath {
 
-							uuid := composite.GetField(inter, interpreter.ReturnEmptyLocationRange, "uuid")
-							if _, ok := uuid.(interpreter.UInt64Value); !ok {
-								log.Printf(
-									"Failed to get UUID for resource @ 0x%x %s",
-									address, key,
-								)
-							}
-						}
+			reader := bytes.NewReader(data)
+			decoder := interpreter.CBORDecMode.NewStreamDecoder(reader)
+			storable, err := interpreter.DecodeStorable(decoder, atree.StorageIDUndefined)
+			if err != nil {
+				log.Printf(
+					"Failed to decode storable @ 0x%x %s: %s (data: %x)",
```

Comment:
not sure if `Printf` function already does this, but do we need to add a newline at the end of the log, here and other places?
```suggestion
					"Failed to decode storable @ 0x%x %s: %s (data: %x)\n",
```